pkg/ansible/operator/operator.go

```diff
@@ -29,78 +29,84 @@ import (
 // to the manager, and finally running the manager.
 func Run(done chan error, mgr manager.Manager, f *flags.AnsibleOperatorFlags, cMap *controllermap.ControllerMap) {
-	watches, err := runner.NewFromWatches(f.WatchesFile)
+	watchesSlice, err := watches.Load(f.WatchesFile)
 	if err != nil {
-		logf.Log.WithName("manager").Error(err, "Failed to get watches")
+		log.Error(err, "Failed to parse watches")
 		done <- err
 		return
 	}
 	rand.Seed(time.Now().Unix())
 	c := signals.SetupSignalHandler()
 
-	for gvk, runner := range watches {
-
-		// if the WORKER_* environment variable is set, use that value.
-		// Otherwise, use the value from the CLI. This is definitely
-		// counter-intuitive but it allows the operator admin adjust the
-		// number of workers based on their cluster resources. While the
-		// author may use the CLI option to specify a suggested
-		// configuration for the operator.
-		maxWorkers := getMaxWorkers(gvk, f.MaxWorkers)
+	for _, watch := range watchesSlice {
+		runner, err := runner.New(watch)
+		if err != nil {
+			log.Error(err, "Failed to create runner")
+			done <- err
+			return
+		}
 
 		o := controller.Options{
-			GVK:          gvk,
-			Runner:       runner,
-			ManageStatus: runner.GetManageStatus(),
-			MaxWorkers:   maxWorkers,
-		}
-		applyFlagsToControllerOptions(f, &o)
-		if d, ok := runner.GetReconcilePeriod(); ok {
-			o.ReconcilePeriod = d
+			GVK:             watch.GroupVersionKind,
+			Runner:          runner,
+			ManageStatus:    watch.ManageStatus,
+			MaxWorkers:      getMaxWorkers(watch.GroupVersionKind, f.MaxWorkers),
+			ReconcilePeriod: watch.ReconcilePeriod,
 		}
+
 		ctr := controller.Add(mgr, o)
 		if ctr == nil {
 			done <- errors.New("failed to add controller")
 			return
 		}
 		cMap.Store(o.GVK, &controllermap.Contents{Controller: *ctr,
-			WatchDependentResources:     runner.GetWatchDependentResources(),
-			WatchClusterScopedResources: runner.GetWatchClusterScopedResources(),
+			WatchDependentResources:     watch.WatchDependentResources,
+			WatchClusterScopedResources: watch.WatchClusterScopedResources,
 			OwnerWatchMap:               controllermap.NewWatchMap(),
 			AnnotationWatchMap:          controllermap.NewWatchMap(),
 		})
 	}
 	done <- mgr.Start(c)
 }
 
-func getMaxWorkers(gvk schema.GroupVersionKind, defvalue int) int {
-	envvar := formatEnvVar(gvk.Kind, gvk.Group)
-	maxWorkers, err := strconv.Atoi(os.Getenv(envvar))
-	if err != nil {
+// if the WORKER_* environment variable is set, use that value.
+// Otherwise, use the value from the CLI. This is definitely
+// counter-intuitive but it allows the operator admin adjust the
+// number of workers based on their cluster resources. While the
+// author may use the CLI option to specify a suggested
+// configuration for the operator.
+func getMaxWorkers(gvk schema.GroupVersionKind, defValue int) int {
+	envVar := strings.ToUpper(strings.Replace(
+		fmt.Sprintf(workerEnvFormat, gvk.Kind, gvk.Group),
```

Comment:
I would've left the format string in the Sprintf. Putting it in a constant really makes this line hard to read what it's doing. Also `workerEnvFormat` doesn't seem to be used anywhere outside of this function.

maybe a comment above the line `envVar := strings....` explaining what is happening.

Reply:
Fair enough. Putting the value back (instead of using a constant) makes adding a comment seem silly. I'll just go back to the format string.